src/Core/Grid/Presenter/AccessbilityChecker/Row/RowAccessibilityCheckerChain.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Presenter\AccessbilityChecker\Row;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionInterface;
+use RuntimeException;
+
+final class RowAccessibilityCheckerChain implements RowAccessibilityCheckerChainInterface
+{
+    /**
+     * @var RowAccessibilityCheckerInterface[]
+     */
+    private $accessibilityCheckers = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check(RowActionInterface $action, array $record, $gridId)
+    {
+        foreach ($this->accessibilityCheckers as $accessibilityChecker) {
+            if ($accessibilityChecker->supports($action, $gridId)) {
+                return $accessibilityChecker->isGranted($record);
```

Comment:
what if the first one say true, and the second one say false? We could have false positive.

Like "only authorized active admin user"

First accessibility checker checks the user is admin => return true, but WUT he's not active anymore! Too late, we will never know :)

Reply:
Be careful, the way it's implemented it's not a Chain, see [Chain of responsibility](https://refactoring.guru/design-patterns/chain-of-responsibility).

Minor improvement: replace `check` by `isGranted` (and I let you refactor ;) ).

Reply:
> what if the first one say true, and the second one say false

it was meant to have 1 checker per action.

Reply:
couldn't this class simply implement `RowAccessibilityCheckerInterface`, I'm not sure having a `RowAccessibilityCheckerChainInterface` is useful, this way `ActionColumnDataPresenter` would simply use a `RowAccessibilityCheckerInterface` as well which lets you use it in more contexts (whether you only have one checker or chained checkers they all have the same interface)